lib/src/rpc/dial.dart

```diff
@@ -0,0 +1,352 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter_webrtc/flutter_webrtc.dart';
+import 'package:grpc/grpc.dart';
+import 'package:grpc/grpc_connection_interface.dart';
+import 'package:logger/logger.dart';
+
+import '../domain/web_rtc/web_rtc_client/web_rtc_client.dart';
+import '../gen/proto/rpc/v1/auth.pb.dart' as pb;
+import '../gen/proto/rpc/v1/auth.pbgrpc.dart';
+import '../gen/proto/rpc/webrtc/v1/signaling.pbgrpc.dart';
+
+final _logger = Logger();
+
+class DialOptions {
+  /// The entity to authenticate as
+  String? authEntity;
+
+  /// Used to authenticate the request.
+  Credentials? credentials;
+
+  /// Options to control how WebRTC is utilized in a dial attempt
+  DialWebRtcOptions? webRtcOptions;
+
+  /// The address to perform external auth at.
+  String? externalAuthAddress;
+
+  /// The entity to authenticate for after externally authenticating.
+  String? externalAuthToEntity;
+
+  /// Allows a pre-authenticated client to dial with
+  /// an authorization header. Direct dial will have the access token
+  /// appended to the "Authorization: Bearer" header. WebRTC dial will
+  /// append it to the signaling server communication
+  ///
+  /// If enabled, other auth options have no affect. Eg. [authEntity], [credentials],
+  /// [externalAuthAddress], [externalAuthToEntity]
+  String? accessToken;
```

Comment:
was there a particular reason you made these Strings`?` nullable?

IMO It adds a lot of boiler plate below to check if they're null and then if they're empty.

could we make them non nullable and initialize them to '' to skip those null checks?

```suggestion
  String authEntity = '';

  /// Used to authenticate the request.
  Credentials? credentials;

  /// Options to control how WebRTC is utilized in a dial attempt
  DialWebRtcOptions? webRtcOptions;

  /// The address to perform external auth at.
  String externalAuthAddress  = '';

  /// The entity to authenticate for after externally authenticating.
  String externalAuthToEntity  = '';

  /// Allows a pre-authenticated client to dial with
  /// an authorization header. Direct dial will have the access token
  /// appended to the "Authorization: Bearer" header. WebRTC dial will
  /// append it to the signaling server communication
  ///
  /// If enabled, other auth options have no affect. Eg. [authEntity], [credentials],
  /// [externalAuthAddress], [externalAuthToEntity]
  String accessToken  = '';
```

Reply:
Decided against doing this (because I liked the nil coalescing later on for convenience), so I instead made an extension on `String?` that will check for both null and empty. Makes things the cleanest :)